### ISSUE TITLE
Need a weighted loss function/ log likelihood

### ISSUE THREAD

**Issue by deveshjawla** [TuringLang/AdvancedVI.jl#44]
For imbalanced datasets a weighted loss function works better than either oversampling or under sampling. Can we add this feature in Turing.jl and AdvancedVI? The log likelihood function would require a weight term added for each class, and according to the number of samples in the training dataset, we would scale the corresponding log likelihood term by the inverse of the number of the samples for that class. I would be happy to contribute but can't find where is it located in the source code, could someone please help?

Weighted Loss: [https://medium.com/gumgum-tech/handling-class-imbalance-by-introducing-sample-weighting-in-the-loss-function-3bdebd8203b4](url)

**Comment by Red-Portal** [comment]
This would belong to `Turing.jl` since it's a model issue than an inference issue. I think it currently can be implemented as follows using `Turing.@addlogprob!`. For example:

```julia
  Turing.@addlogprob! mapreduce(+, 1:n_datapoint) do n
      w[n]*Turing.Turing.BernoulliLogit(logit[n])
  end.
```

Please re-open if this does not resolve your problem.
